code/__HELPERS/botany.dm

```diff
@@ -0,0 +1,36 @@
+/// Tray Setters - The following macros adjust the hydroponics tray variables, and make sure that the stat doesn't go out of bounds.
+/**
+ * Adjust water.
+ * Raises or lowers tray water values by a set value. Adding water will dillute toxicity from the tray.
+ * * adjustamt - determines how much water the tray will be adjusted upwards or downwards.
+ */
+#define TRAY_ADJUST_WATER(tray, amt) tray.set_waterlevel(clamp(tray.waterlevel + amt, 0, tray.maxwater), FALSE)
+
+/**
+ * Adjust Health.
+ * Raises the tray's plant_health stat by a given amount, with total health determined by the seed's endurance.
+ * * adjustamt - Determines how much the plant_health will be adjusted upwards or downwards.
+ */
+#define TRAY_ADJUST_HEALTH(tray, amt) tray.set_plant_health(clamp(tray.plant_health + amt, 0, tray.myseed?.endurance), FALSE)
```

Comment:
What will happen if `tray.myseed` is null?

Reply:
the plant health will be set to zero. which is what happens when there's no plant anyway.